flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/typeutils/LinkedListSerializer.java

```diff
@@ -110,17 +124,40 @@ public int getLength() {
     @Override
     public void serialize(LinkedList<T> list, DataOutputView target) throws IOException {
         target.writeInt(list.size());
+        if (hasNullMask) {
+            MaskUtils.writeMask(getNullMask(list), target);
+        }
         for (T element : list) {
-            elementSerializer.serialize(element, target);
+            if (element != null) {
+                elementSerializer.serialize(element, target);
+            }
         }
     }
 
+    private boolean[] getNullMask(LinkedList<T> list) {
+        boolean[] mask = new boolean[list.size()];
```

Comment:
I think we should reuse this mask.

Reply:
I suppose by "reusing" you're meant to not creating a boolean array every time and change its size when the length of list grows.